src/cpp/StatisticsBackend.cpp

```diff
@@ -182,34 +180,44 @@ EntityId create_and_register_monitor(
         const DomainParticipantQos& participant_qos,
         const DomainId domain_id = 0)
 {
+    // NOTE: This method is quite awful to read because of the error handle of every entity
```

Comment:
This method uses the internal namespace a lot, so using it will improve readability.
It would also be nice to call `StatisticsBackendData::get_instance()` once at the beginning.

```suggestion
{
    using namespace details;
    auto backend_data = StatisticsBackendData::get_instance();

    // NOTE: This method is quite awful to read because of the error handle of every entity
```